src/code/code_800EC960.c

```diff
@@ -4649,52 +4649,47 @@ void func_800F574C(f32 scaleTempoAndFreq, u8 duration) {
     SEQCMD_SETUP_SET_PLAYER_FREQ(SEQ_PLAYER_FANFARE, SEQ_PLAYER_BGM_MAIN, duration, scaleTempoAndFreq * 100.0f);
 }
 
-void func_800F5918(void) {
-    if (Audio_GetActiveSeqId(SEQ_PLAYER_BGM_MAIN) == NA_BGM_TIMED_MINI_GAME &&
+void Audio_IncreaseTempoForTimedMinigame(void) {
+    if ((Audio_GetActiveSeqId(SEQ_PLAYER_BGM_MAIN) == NA_BGM_TIMED_MINI_GAME) &&
         Audio_IsSeqCmdNotQueued(SEQCMD_OP_PLAY_SEQUENCE << 28, SEQCMD_OP_MASK)) {
         SEQCMD_SET_TEMPO(SEQ_PLAYER_BGM_MAIN, 5, 210);
     }
 }
 
-void func_800F595C(u16 arg0) {
-    u8 arg0b = arg0 & 0xFF;
-
-    if (sSeqFlags[arg0b] & SEQ_FLAG_FANFARE) {
-        Audio_PlayFanfare(arg0);
-    } else if (sSeqFlags[arg0b] & SEQ_FLAG_FANFARE_GANON) {
-        SEQCMD_PLAY_SEQUENCE(SEQ_PLAYER_FANFARE, 0, 0, arg0);
-
+void Audio_PlaySequenceInCutscene(u16 seqId) {
+    if (sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE) {
+        Audio_PlayFanfare(seqId);
+    } else if (sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE_GANON) {
+        SEQCMD_PLAY_SEQUENCE(SEQ_PLAYER_FANFARE, 0, 0, seqId);
     } else {
-        Audio_PlaySequenceWithSeqPlayerIO(SEQ_PLAYER_BGM_MAIN, arg0, 0, 7, SEQ_IO_VAL_NONE);
+        Audio_PlaySequenceWithSeqPlayerIO(SEQ_PLAYER_BGM_MAIN, seqId, 0, 7, SEQ_IO_VAL_NONE);
         SEQCMD_STOP_SEQUENCE(SEQ_PLAYER_FANFARE, 0);
     }
 }
 
-void func_800F59E8(u16 arg0) {
-    u8 arg0b = arg0 & 0xFF;
-
-    if (sSeqFlags[arg0b] & SEQ_FLAG_FANFARE) {
+void Audio_StopSequenceInCutscene(u16 seqId) {
+    if (sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE) {
```

Comment:
my theory is there were these macros
```c
#define SEQ_ID(packedSeqId) ((packedSeqId) & 0xFF)
#define SEQ_FLAGS(packedSeqId) sSeqFlags[SEQ_ID(packedSeqId)]
```
and
```suggestion
    if (SEQ_FLAGS(SEQ_ID(seqId)) & SEQ_FLAG_FANFARE) {
```
I haven't properly researched this possibility though, but if this can work then this PR may be a good opportunity to implement that too

Reply:
What you have makes sense to me since the `seqId` often contains both the `SeqId` enum (it should be converted from defines to an enum at some point), but also contains `seqArgs` from `Audio_StartSequence`, so the `& 0xFF` to extract the index is logical. We need a name to capture both `seqId` and `seqArgs` in 1 variable. `packedSeqId` as you have or `flaggedSeqId` or something else.

To summarize usage, `sSeqFlags` is used 14 times:
```c
(sSeqFlags[sPrevSceneSeqId] & SEQ_FLAG_RESUME_PREV) // can not have any `& 0xFF`
(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_RESUME)
(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_SKIP_HARP_INTRO)
(sSeqFlags[seqId] & SEQ_FLAG_RESUME_PREV)
(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_RESUME)
(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE)
(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE_GANON)
(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE)
(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE_GANON)
(sSeqFlags[seqId & 0xFF] & SEQ_FLAG_FANFARE)
(sSeqFlags[seqId & 0xFF] & SEQ_FLAG_FANFARE_GANON)
(sSeqFlags[Audio_GetActiveSeqId(SEQ_PLAYER_BGM_MAIN) & 0xFF] & SEQ_FLAG_RESTORE)
(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_ENEMY)
(sSeqFlags[gActiveSeqs[SEQ_PLAYER_BGM_MAIN].seqId & 0xFF & 0xFF] & SEQ_FLAG_NO_AMBIENCE)
```

So:
9/14 -> use the `& 0xFF & 0xFF` pattern
3/14 -> use the `& 0xFF` pattern
2/14 -> uses no extra masks

So it's inconsistent? Seems like the ones with missing masks don't want them in terms of matching

Reply:
Note one of those seqId s is a u8 (`sPrevSceneSeqId`)

Function|index|access
--|--|--
`Audio_PlaySceneSequence`|`u8 sPrevSceneSeqId`|`(sSeqFlags[sPrevSceneSeqId] & SEQ_FLAG_RESUME_PREV) // can not have any & 0xFF`
`Audio_PlaySceneSequence`|`u16 seqId`|`(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_RESUME)`
`Audio_PlaySceneSequence`|`u16 seqId`|`(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_SKIP_HARP_INTRO)`
`Audio_PlaySceneSequence`|`u16 seqId`|`(sSeqFlags[seqId] & SEQ_FLAG_RESUME_PREV)`
`Audio_UpdateSceneSequenceResumePoint`|`u16 seqId`|`(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_RESUME)`
`Audio_PlaySequenceInCutscene`|`u16 seqId`|`(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE)`
`Audio_PlaySequenceInCutscene`|`u16 seqId`|`(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE_GANON)`
`Audio_StopSequenceInCutscene`|`u16 seqId`|`(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE)`
`Audio_StopSequenceInCutscene`|`u16 seqId`|`(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_FANFARE_GANON)`
`Audio_IsSequencePlaying`|`u8 seqId`|`(sSeqFlags[seqId & 0xFF] & SEQ_FLAG_FANFARE)`
`Audio_IsSequencePlaying`|`u8 seqId`|`(sSeqFlags[seqId & 0xFF] & SEQ_FLAG_FANFARE_GANON)`
`func_800F5B58`|`u16 Audio_GetActiveSeqId(...)`|`(sSeqFlags[Audio_GetActiveSeqId(SEQ_PLAYER_BGM_MAIN) & 0xFF] & SEQ_FLAG_RESTORE)`
`Audio_SetSequenceMode`|`u16 seqId`|`(sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_ENEMY)`
`Audio_PlayNatureAmbienceSequence`|`u16 ActiveSequence.seqId`|`(sSeqFlags[gActiveSeqs[SEQ_PLAYER_BGM_MAIN].seqId & 0xFF & 0xFF] & SEQ_FLAG_NO_AMBIENCE)`

The only really weird one in the end is the fourth entry
`Audio_PlaySceneSequence`, `u16 seqId`, `(sSeqFlags[seqId] & SEQ_FLAG_RESUME_PREV)`
which indexes `sSeqFlags` with a u16 (all other occurrences are basically u8).

And yeah I think the usage was inconsistent between `SEQ_FLAGS(SEQ_ID(seqId))` and `SEQ_FLAGS(seqId)`, and/or the `SEQ_FLAGS` macro changed during development (like, the inner `SEQ_ID` would have been added later on)
Assuming this was a macro at all, but imho that's quite likely

Reply:
I was able to turn the arg from `Audio_IsSequencePlaying` from u8 to u16 to match the pattern of the other functions and still get a match. I've incorporated the macros but not sure if they need additional comments or not

Reply:
See discussion above, I'll be leaving these macros to a separate PR